proxy/envoy/config.go

```diff
@@ -163,6 +180,43 @@ func build(instances []*model.ServiceInstance, services []*model.Service,
 				},
 			}},
 		}
+
+		// Add TCP filters that share a port with HTTP filters.
+		if _, ok := tcpRouteConfigs[port]; ok {
+
+			// TCP and HTTP filter on the same port seems to cause
+			// envoy to assign all traffic to TCP filter regardless of
+			// the routing config. Temporarily drop TCP filters that
+			// share a port with HTTP (e.g. istio-egress) until this is
+			// root caused.
+
```

Comment:
@rshriram, re: your original comment, did you mean that we need to the generate the equivalent TCPProxy filter for each HTTP service? 

@kuat - yes, A filter seems to catch all traffic for the listener port regardless of the dst/src in the TCP route config. If a TCP Proxy filter is specified for a port, then the corresponding TCP route needs to match, otherwise the filter returns StopIteration and no additional filters are checked (and the traffic isn't forwarded). I didn't find a way to configure envoy to skip over the TCP proxy filter if dst/src didn't match and fallback to the HTTP connection pool mechanism.